designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala

```diff
@@ -490,11 +491,12 @@ class AkkaHttpBasedRouteProvider(
         designerClock
       )
 
-      val statisticUrlConfig = if (usageStatisticsReportsConfig.encryptionEnabled) {
-        StatisticUrlConfig(maybePublicEncryptionKey = Some(PublicEncryptionKey.INSTANCE))
-      } else {
-        StatisticUrlConfig()
-      }
+      val publicKey =
```

Comment:
I would move this logic after line 143:
```
statisticsPublicKey <- Resource.fromAutoCloseable(
        IO {
          Source.fromURL(getClass.getResource("/encryption.key"))
        }
      )
```
thanks to this we will use cats Resource and errors will be handled in proper way. What do you think?